### PR TITLE
Dev/deezer

### PR DESCRIPTION
**Pull request by IkkyOdufade**
1. added deezer as a music service.
2. added an alert for non-registered users but having issues because i used window.alert().  Can you give me an alternative? I don't want it to throw an error. 

Before submitting a final PR, please:

- [ ] Run `npm test` on your machine
- [ ] Run `npm run format` to keep code formatted consistently

### CODE REVIEW

**Comment by IkkyOdufade** [comment]
Waiting on your feedback @ianb. Thanks.

File: extension/services/deezer/deezer.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import * as music from "../../intents/music/music.js";
+import * as serviceList from "../../background/serviceList.js";
+import * as content from "../../background/content.js";
+import { shouldDisplayWarning } from "../../limiter.js";
+
+class Deezer extends serviceList.Service {
```
Comment (ianb) [inline extension/services/deezer/deezer.js]: You should be able to reduce this code if you use `import MusicService from "../../background/musicService.js";` like `soundcloud.js`

File: extension/services/deezer/deezer.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  async playQuery(query) {
+    await this.initTab("/services/deezer/player.js");
+    if(this.initTab){
+   window.alert ("Please signup or login to use this service. If you have, please ignore this message.");
```
Comment (ianb) [inline extension/services/deezer/deezer.js]: I think you can do:

```js
this.context.activateTab(this.tab);
const exc = new Error("Not logged in");
exc.displayMessage = "Please log in";
throw exc;
```

That said, this probably isn't the right place to raise this exception. The tab might get opened but the user might still be logged in. Instead you should let `player.js` raise the exception and catch it

File: extension/services/deezer/deezer.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    }
+    await this.callTab("search", { query, thenPlay: true });
```
Comment (ianb) [inline extension/services/deezer/deezer.js]: So here there should be something like:

```
try {
  await this.callTab(...);
} catch (e) {
  if (e.message === "Not logged in") {
    e.displayMessage = "Please log in";
    this.context.activateTab(this.tab);
  }
  throw e;
}
```

And maybe this should go in `musicPlayer.js`

File: extension/services/deezer/player.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
+/* globals helpers */
+
+this.player = (function() {
+  const SEARCH_PLAY = ".datagrid-row song a[role='button']";
+
+  class Player extends helpers.Runner {
+    action_play() {
+      const button = this.querySelector("button[aria-label='Play']");
```
Comment (ianb) [inline extension/services/deezer/player.js]: I realize spotify works like this currently, but if it's possible to not use the labels that would be better: then if someone is using a localized version of the site, these selectors will still work.

File: README.md [hunk 5]
```diff
@@ -91,24 +91,6 @@ You will need to setup Firefox Nightly or Developer on WSL before running `firef
...
-Before cloning the repo, using the terminal in Git-bash:
-
-1. Run `git config --global core.autocrlf false` to prevent git from converting line endings from LF to CRLF
-2. Run `npm config set script-shell "path\\to\\bash.exe"`. Example of path is `"C:\\Program Files\\git\\bin\\bash.exe"`. This enables npm to run linux-like commands.
-3. Clone the [repository](https://github.com/mozilla/firefox-voice.git) `git clone https://github.com/mozilla/firefox-voice.git`
-4. To create the environment variable `FIREFOX`, run `export FIREFOX=normalized/path/to/firefox.exe`. For example if the install path is `C:\Program Files\Firefox Nightly\firefox.exe` then normalized path is `/c/Program Files/Firefox Nightly/firefox.exe`. Alternatively, the environment variable set to the normalized path can be created using the Windows system dialog.
-5. Now run `npm install`
-6. Run `npm start`. This will launch a new Firefox browser with the `firefox-voice` extension installed.
-
-
```
Comment (ianb) [inline README.md]: I'm guessing you didn't mean to include this change in the PR...?

**Comment by IkkyOdufade** [comment]
Ok. Thank you.

File: README.md (hunk 5 above)
Comment (IkkyOdufade) [reply README.md]: I didnt....

**Comment by IkkyOdufade** [comment]
Hi @ianb . Please go through my corrections and I'll be waiting for your feedback. Thank you.

File: README.md (hunk 5 above)
Comment (ianb) [reply README.md]: I think if you run `git checkout origin/master -- README.md` it should undo this.

File: extension/background/MusicService.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
+import * as serviceList from "../../background/serviceList.js";
```
Comment (ianb) [inline extension/background/MusicService.js]: This should be using the existing file in `extension/background/musicService.js` (if you don't see that file you'll need to merge master into your branch/fork)

Comment (IkkyOdufade) [reply extension/background/MusicService.js]: Hi @ianb . Thank you for your corrections. Would work on it now.

**Comment by IkkyOdufade** [comment]
Hi Ian. I had to close this pr because I could not get git to do what I wanted. So I created a new branch and opened a new pr, with just deezer. Thank you.

**Pull request closed without merging**